### ISSUE TITLE
Truncated titles

### ISSUE THREAD

**Issue by DavidGarciaCat** [sonata-project/SonataAdminBundle#4861]
Hello,

Is there a reason to truncate titles?

https://github.com/sonata-project/SonataAdminBundle/blob/3.x/src/Resources/views/CRUD/base_show.html.twig#L14-L16

```php
{% block title %}
    {{ "title_show"|trans({'%name%': admin.toString(object)|truncate(15) }, 'SonataAdminBundle') }}
{% endblock %}
```

I have been requested to display the title as it's provided, however looks like this Twig conditional is not allowing parameters to truncate the title or leave it as it is returned by the `__toString` magic method.

Could be possible to add a parameter to allow users to keep the title as it's returned by the method?

**Comment by OskarStark** [comment]
I am totally with you and I like this "feature" neither.

@sonata-project/contributors WDYT, shall we change this behavior to show the complete title or make in configurable?

**Comment by jordisala1991** [comment]
You should be able to override the template and change that specific block.

Truncate is there to avoid really long names. I dont think we want more options to change that, if we already have a way to override it.

**Comment by OskarStark** [comment]
> Truncate is there to avoid really long names.

ok, but 15 chars is very short, maybe we could use 50 or sth like this

**Comment by jordisala1991** [comment]
It depends. If you have multiple tabs on the admin or you are using lower resolution, You might be needing not a lot of chars there.

**Comment by OskarStark** [comment]
But couldn't we truncate it via css?

![screenshot 2017-12-21 08 14 00](https://user-images.githubusercontent.com/995707/34244714-f2a4f6da-e626-11e7-8dbd-cf888bd9d61d.png)

**Comment by DavidGarciaCat** [comment]
IMHO @OskarStark 's proposal to truncate it via CSS makes more senes to me

As @jordisala1991 pointed, screen size could be a problem, but we build responsive sites, right? So this truncated copy could be managed via CSS based on the screen resolution.

**Comment by core23** [comment]
The title is also used for the browser title, so we should truncate it at the server.

**Comment by DavidGarciaCat** [comment]
Hey @core23 

Why to truncate the browser title? In the end, Sonata Admin provides and admin panel that - by definition - requires a valid login, so SEO and other stuff that could be affected by this is not a problem.

**Comment by OskarStark** [comment]
> The title is also used for the browser title, so we should truncate it at the server.

why do we have 2 titles then?

https://github.com/sonata-project/SonataAdminBundle/blob/3.x/src/Resources/views/CRUD/base_show.html.twig#L14-L20

**Comment by core23** [comment]
The `title` is used for the browser:
https://github.com/sonata-project/SonataAdminBundle/blob/3.x/src/Resources/views/standard_layout.html.twig#L93-L95

The `navbar_title` (surprisingly) for the navbar:
https://github.com/sonata-project/SonataAdminBundle/blob/3.x/src/Resources/views/standard_layout.html.twig#L278-L280

**Comment by core23** [comment]
> Why to truncate the browser title?

Because an entity with a veeeeeerrry long name could destroy the browser title so you only get the entity name in the title and no further information which side you are on.

**Comment by DavidGarciaCat** [comment]
Hey @core23 

Based on W3C title should be okay as far as length is 64 characters:
https://www.w3.org/Provider/Style/TITLE.html

SEO Moz suggests between 50-60 however I guess W3C is more relevant here:
https://moz.com/learn/seo/title-tag

And SEO Quake through their browser extension says up to 70 characters is fine:
_"The most effective page titles are about 10-70 characters long, including spaces."_

I don't think that truncate the title to display just 15 characters is the best here, based on these reports from W3C and SEO Tools, so maybe allow between 50 and 70 could be better.

Thoughts?

**Comment by core23** [comment]
Please be aware, that the full title is not cropped to 15 chars. Only the entity name is cropped to 15 chars. 

Depending on the locale, the title could be something like:

```
Show "My cool entity ..."

"My cool entity ..." betrachten
```

The full title of 50 chars+ would only be visible if you have only one tab open. That's the reason why we have the limit. 

Feel free to override this in your project template.

**Comment by DavidGarciaCat** [comment]
Hi,

Just to confirm, before to close the issue: I just need to extend a new template form the base one and then set it on the project settings, like this?

`app/config/config.yml`
```yaml
sonata_admin:
    templates:
        show: ::sonata_show.html.twig
```

`app/Resources/views/sonata_show.html.twig`
```php
{% extends 'SonataAdminBundle:CRUD:base_show.html.twig' %}

{% block title %}
    {{ "title_show"|trans({'%name%': admin.toString(object) }, 'SonataAdminBundle') }}
{% endblock %}
```

I have tested this, but I just want to make sure - from your POV - if I should add any other change that I could be missing.

Thanks,

**Comment by DavidGarciaCat** [comment]
About that example:
```
Show "My cool entity ..."
"My cool entity ..." betrachten
```
Is a potential update to do something like this? `[action]: [title]` so:
```
Show: "My cool entity..."
Betrachten: "My cool ..."
```
In this way we can see always the "action" that we are doing, and we can truncate it at 60 characters:
```
{{ block('title')|truncate(60) }}
```
Does this make sense?

**Comment by phansys** [comment]
I'm closing this issue since #Closes #5595 is already closed and released on version [3.5.0](https://github.com/sonata-project/SonataAdminBundle/releases/tag/3.50.0).
Thank you!
